src/logic/voice/voice.ts

```diff
@@ -73,10 +78,100 @@ export function createVoiceComponent(
     await voiceDB.updateUserStatusAsConnectionInterrupted(userAddress, roomName)
   }
 
+  /**
+   * Handles the event when a participant joins a COMMUNITY voice chat room.
+   */
+  async function handleCommunityParticipantJoined(userAddress: string, roomName: string): Promise<void> {
+    logger.debug(`Community participant joined: ${userAddress} in room ${roomName}`)
+    // Simply update the user status to connected
+    await voiceDB.updateCommunityUserStatus(userAddress, roomName, VoiceChatUserStatus.Connected)
+  }
+
+  /**
+   * Handles the event when a participant leaves a COMMUNITY voice chat room.
+   */
+  async function handleCommunityParticipantLeft(
+    userAddress: string,
+    roomName: string,
+    disconnectReason: DisconnectReason
+  ): Promise<void> {
+    logger.debug(`Community participant left: ${userAddress} in room ${roomName}, reason: ${disconnectReason}`)
+
+    if (disconnectReason === DisconnectReason.DUPLICATE_IDENTITY) {
+      logger.debug('Ignoring disconnect due to duplicate identity')
+      return
+    }
+
+    if (disconnectReason === DisconnectReason.ROOM_DELETED) {
+      await voiceDB.deleteCommunityVoiceChat(roomName)
+      return
+    }
+
+    // Update user status based on disconnect reason
+    if (disconnectReason === DisconnectReason.CLIENT_INITIATED) {
+      await voiceDB.updateCommunityUserStatus(userAddress, roomName, VoiceChatUserStatus.Disconnected)
+
+      // Only check for room destruction when a moderator leaves voluntarily
+      // Get all users to check if the leaving user was a moderator
+      const usersInRoom = await voiceDB.getCommunityUsersInRoom(roomName)
+      const leavingUser = usersInRoom.find((user) => user.address === userAddress)
+
+      if (leavingUser?.isModerator) {
+        logger.debug(`Moderator ${userAddress} left voluntarily, checking if room should be destroyed`)
+
+        // Check if there are any other active moderators
+        const activeModerators = usersInRoom.filter(
```

Comment:
A moderator is considered active if he's connected or if he his connection has been interrupted or if he has never connected to the room but the TTL has not passed.
If I'm not wrong, the `shouldDestroyCommunityRoom` might be useful to compute this (or maybe the logic can be extracted in some way it's useful) as we're kind of needing the same here.